pyvo/mivot/seekers/resource_seeker.py

```diff
@@ -0,0 +1,118 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Class that provides multiple getters on VOTable RESOURCE elements.
+"""
+from pyvo.mivot.utils.vocabulary import Constant
+from pyvo.utils.prototype import prototype_feature
+
+
+@prototype_feature('MIVOT')
+class ResourceSeeker:
+    """
+    This class provides multiple getters on resource tables.
+    Some methods are simple wrappers for external tools in order to have all
+    the search functions on RESOURCE gathered in within a single namespace.
+    """
+    def __init__(self, resource):
+        """
+        Constructor
+        Parameters
+        ----------
+        resource (astropy.votable.Resource): The resource object to be queried.
+        """
+        self._resource = resource
+
+    def get_table_ids(self):
+        """
+        Return the list of table ids.
+        Only resource children are considered.
+        The @ID is first searched and then the @name, and finally 'AnonymousTable' is taken.
+        Returns
+        -------
+        list of str: table ids.
+        """
+        ids_found = []
+        for table in self._resource.tables:
+            if table.ID is not None:
+                ids_found.append(table.ID)
+            elif table.name is not None:
+                ids_found.append(table.name)
+            else:
+                ids_found.append(Constant.ANONYMOUS_TABLE)
+        return ids_found
+
+    def get_table(self, table_name):
+        """
+        Return the table matching table_name first by ID and then by name.
+        Parameters
+        ----------
+        table_name (str): Name of the table to get.
+        Returns
+        -------
+        ~astropy.votable.table: table matching the table_name.
+        """
+        if table_name == Constant.FIRST_TABLE:
+            return self._resource.tables[0]
+        for table in self._resource.tables:
+            if (table_name is None or table.name == table_name
+                    or table.ID == table_name):
+                return table
+        return None
+
+    def get_params(self):
+        """
+        Return the VOTable PARAMS.
+        Returns
+        -------
+        ~astropy.votable.Resource.params: VOTable PARAMS.
+        """
+        return self._resource.params
+
+    def get_id_index_mapping(self, table_name):
+        """
+        Build an index binding column number with field id.
+        Parameters
+        ----------
+        table_name (str): Name of the table.
+        Returns
+        -------
+        dict: dictionary mapping field id to column number: {name: {ID, ref, indx}...}
```

Comment:
This is a bit wired, but there are good reasons for it. 
- The `ATTRIBUTE@ref` can refer to either the `FIELD@name` or the `FIELD@ID`. So we need to keep track of both identifiers to resolve ATTRIBUTE.
- We also prefer to keep track of the column index, because some users may apply the parsed mapping block to data tables not provided by a VOTable parser. 

Therefore, packing fields in a dictionary seemed to be the most appropriate way to proceed.